File: librarymanager/librarymanager-be/src/main/java/ru/spring5/elena/librarymanager/domain/entity/EditionEntity.java
package ru.spring5.elena.librarymanager.domain.entity;

import lombok.*;
import ru.spring5.elena.librarymanager.util.AgeRating;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.HashSet;
import java.util.Set;

import static javax.persistence.GenerationType.IDENTITY;

@Entity
@Table(name = "EDITIONS")
@Getter
@NoArgsConstructor
//@RequiredArgsConstructor
@Setter
public class EditionEntity {

    @Id
    @GeneratedValue(strategy = IDENTITY) //уникальный ID, генерируется сам в БД
    @Column(name = "EDITION_ID")
    @NotNull
    @Setter(AccessLevel.NONE)
    private Long editionId;

    @Version
    @Column(name = "VERSION")
    private int version; //версия, инкрементится при редактировании записи

    @Column(name = "TITLE")
    private String title;

    @Column(name = "WORK_TITLE")
    private String workTitle; //

    @Column(name = "YEAR")
    private int year;

    @ManyToOne
    @JoinColumn(name = "GENERE_ID")
    private GenreEntity genre;

    @ManyToOne
    @JoinColumn(name = "LANG_ID")
    private LangEntity lang;

    @ManyToOne
    @JoinColumn(name = "SECTION_ID")
    private SectionEntity section;

    @Column(name = "PUBLISHING_YEAR")
    private int publishingYear;

    @Column(name = "ISBN")
    private String isbn;

    @Column(name = "IS_ILLUSTRATED")
    private boolean isIllustrated;

    @Column(name = "DOWNLOAD_LINK")
    private String downloadLink;

    @Enumerated(EnumType.ORDINAL)
    @Column(name = "AGE")
    private AgeRating age;

    @Column(name = "INFO")
    private String info;

    @ManyToOne
    @JoinColumn(name = "PUBLISHING_HOUSE_ID")
    private PublishingHouseEntity publishingHouse;

    @ManyToOne
    @JoinColumn(name = "PUPL_LANG_ID")
    private LangEntity publLang;

    @ManyToMany
    @JoinTable(name = "AUTHOR_EDITION", joinColumns = @JoinColumn(name = "EDITION_ID"), inverseJoinColumns = @JoinColumn(name = "AUTHOR_ID"))
    @Setter(AccessLevel.NONE)
    private Set<AuthorEntity> authors = new HashSet<AuthorEntity>();

    @OneToMany(mappedBy = "edition", cascade = CascadeType.ALL, orphanRemoval = true)
    @Setter(AccessLevel.NONE)
    private Set<OrderEntity> orders = new HashSet<OrderEntity>();
    
    @OneToMany(mappedBy = "edition", cascade = CascadeType.ALL, orphanRemoval = true)
    @Setter(AccessLevel.NONE)
    private Set<BookEntity> books = new HashSet<BookEntity>();
}

File: librarymanager-be/src/main/kotlin/ru/elenacherev/librarymanager/domain/entity/EditionEntity.kt
package ru.elenacherev.librarymanager.domain.entity

import ru.elenacherev.librarymanager.api.enum.AgeRating
import javax.persistence.CascadeType
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.EnumType
import javax.persistence.Enumerated
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.JoinColumn
import javax.persistence.ManyToMany
import javax.persistence.ManyToOne
import javax.persistence.OneToMany
import javax.persistence.Table
import javax.persistence.Version
import javax.validation.constraints.NotNull

@Entity
@Table(name = "EDITIONS")
class EditionEntity (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY) //уникальный ID, генерируется сам в БД
    @Column(name = "EDITION_ID")
    @get: NotNull
    var editionId:  Long? = null,

    //версия, инкрементится при редактировании записи
    @Version
    @Column(name = "VERSION")
    var version: Int  = 0,

    @Column(name = "TITLE")
    var title: String,

    @Column(name = "WORK_TITLE")
    var workTitle: String,

    @Column(name = "YEAR")
    var year: Int  = 0,

    @ManyToOne
    @JoinColumn(name = "GENERE_ID")
    var genre: GenreEntity,

    @ManyToOne
    @JoinColumn(name = "LANG_ID")
    var lang: LangEntity,

    @ManyToOne
    @JoinColumn(name = "SECTION_ID")
    var section: SectionEntity,

    @Column(name = "PUBLISHING_YEAR")
    var publishingYear: Int = 0,

    @Column(name = "ISBN")
    var isbn: String,

    @Column(name = "IS_ILLUSTRATED")
    var isIllustrated: Boolean = false,

    @Column(name = "DOWNLOAD_LINK")
    var downloadLink: String? = null,

    @Enumerated(EnumType.ORDINAL)
    @Column(name = "AGE")
    var age: AgeRating,

    @Column(name = "INFO")
    var info: String? = null,

    @ManyToOne
    @JoinColumn(name = "PUBLISHING_HOUSE_ID")
    var publishingHouse: PublishingHouseEntity,

    @ManyToOne
    @JoinColumn(name = "PUBL_LANG_ID")
    var publLang: LangEntity,

    @ManyToMany(mappedBy = "editions", targetEntity = AuthorEntity::class) //  @JoinTable(name = "AUTHOR_EDITION", joinColumns = @JoinColumn(name = "EDITION_ID"), inverseJoinColumns = @JoinColumn(name = "AUTHOR_ID"))
    val authors: MutableSet<AuthorEntity> = mutableSetOf(),

    @OneToMany(mappedBy = "edition", cascade = [CascadeType.ALL], orphanRemoval = true)
    val orders: MutableSet<OrderEntity> = mutableSetOf(),

    @OneToMany(mappedBy = "edition", cascade = [CascadeType.ALL], orphanRemoval = true)
    val books: MutableSet<BookEntity> = mutableSetOf()
)